### PR TITLE
TEST-#5280: Test pandas objects for non-commutative multiply.

### LINKED ISSUE #5280: TEST: Test pandas objects instead of modin objects in test_non_commutative_multiply_pandas
[here](https://github.com/modin-project/modin/blob/b103735f05bc5cda2b19bf84f404e95b0bb894d5/modin/pandas/test/dataframe/test_binary.py#L359) and [here](https://github.com/modin-project/modin/blob/b103735f05bc5cda2b19bf84f404e95b0bb894d5/modin/pandas/test/test_series.py#L4281). Based on comment [here](https://github.com/modin-project/modin/pull/5246/files#r1034467366)

### PR DESCRIPTION
**Pull request by mvashishtha**
Signed-off-by: mvashishtha <mahesh@ponder.io>

## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5280 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/pandas/test/dataframe/test_binary.py [hunk 1]
```diff
@@ -360,7 +360,7 @@ def test_non_commutative_multiply_pandas():
     # The non commutative integer class implementation is tricky. Check that
     # multiplying such an integer with a pandas dataframe is really not
     # commutative.
-    pandas_df = pd.DataFrame([[1]], dtype=int)
+    pandas_df = pandas.DataFrame([[1]], dtype=int)
```
Comment (YarShev) [inline modin/pandas/test/dataframe/test_binary.py:363]: Why do we need to test pandas? Doesn't pandas itself do this?

File: modin/pandas/test/test_series.py [hunk 2]
```diff
@@ -4282,7 +4282,7 @@ def test_non_commutative_multiply_pandas():
     # The non commutative integer class implementation is tricky. Check that
     # multiplying such an integer with a pandas series is really not
     # commutative.
-    pandas_series = pd.DataFrame([[1]], dtype=int)
+    pandas_series = pandas.Series(1, dtype=int)
```
Comment (YarShev) [inline modin/pandas/test/test_series.py:4285]: same

File: modin/pandas/test/dataframe/test_binary.py (hunk 1 above)
Comment (mvashishtha) [reply modin/pandas/test/dataframe/test_binary.py:363]: See the comment just above. Getting the non-commutative multiply class right was tricky. Before we check that the multiply with Modin matches the multiply with pandas, we should check that the multiply with pandas is really non-commutative. If it's not, there's no point in running `test_non_commutative_multiply`.

File: modin/pandas/test/test_series.py (hunk 2 above)
Comment (mvashishtha) [reply modin/pandas/test/test_series.py:4285]: ditto

**Review by YarShev — APPROVED** [review]
@mvashishtha, thanks!

**Pull request merged**